bpf/bpf_lxc.c

```diff
@@ -1023,8 +1030,9 @@ ipv4_policy(struct __sk_buff *skb, int ifindex, __u32 src_label, __u8 *reason, _
 	// Do not redirect again if the packet is coming from the egress proxy.
 	if ((ret == CT_REPLY || ret == CT_RELATED) && ct_state.proxy_redirect &&
 	    !tc_index_skip_egress_proxy(skb)) {
-		// Stack will do a socket match and deliver locally
-		return skb_redirect_to_proxy(skb, 0);
+		send_trace_notify(skb, TRACE_TO_PROXY, src_label, SECLABEL,
+				  0, ifindex, *reason, monitor);
+		return TC_ACT_PROXY_REDIRECT;
```

Comment:
The implicit set of `*proxy_port` to zero here is a little surprising, IMO that should be explicitly called out somehow.

Reply:
I don't quite follow this comment. Are you talking about the trace message or the caller of `ipv4_policy()` still redirecting?

Reply:
I was looking at the previous equivalent code, `skb_redirect_to_proxy(..., 0)` where the port is set to zero because the stack already knows that it needs to receive packets for this tuple and hence we don't need to embed the port into the `skb->mark` (which would otherwise match the tproxy rules and then zero the port portion of the mark).

This kind of detail where this path relies upon passing the port back as zero seems really easy to miss if we subsequently refactor.